### PR TITLE
Sjuls/slack 2

### PR DESCRIPTION
**Pull request by sjuls**

### CODE REVIEW

File: dbctx/soup.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+// NewSoupRepository creates a new repository
+func NewSoupRepository(connFactory ConnectionFactory) SoupRepository {
+	return &soupRepository{
+		connFactory,
+	}
+}
+
+func (m *soupRepository) SetSoup(name string) error {
```
Comment (mbang-mlyfe) [inline dbctx/soup.go:28]: Consider renaming this to `SetSoupOfTheDay`

Comment (sjuls) [reply dbctx/soup.go:28]: Yeah - good point

File: dbctx/soup.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	defer db.Close()
+
+	tx := db.Begin()
+	soupOfTheDay := getCreateSoupOfTheDay(tx)
+	tx.Commit()
+	return soupOfTheDay, tx.Error
+}
+
+// getCreateSoupOfTheDay returns the soup of the day, pass DB or transaction to be used to extract the data.
+func getCreateSoupOfTheDay(tx *gorm.DB) *SoupOfTheDay {
```
Comment (mbang-mlyfe) [inline dbctx/soup.go:60]: Is it on purpose that you don't have this as a method on the struct? I'm considering whether there's a use-case for not doing so. The only thing I can come up with right now is that other parts of the package could call `getCreateSoupOfTheDay` without having a `soupRepository`

File: slack/commands/rate.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	rateMatches := rateMatcher.FindStringSubmatch(args)
+	scoreValue, err := strconv.ParseInt(rateMatches[1], 10, 32)
+
+	if err != nil {
+		fmt.Fprintln(output, err.Error())
+		return
+	}
+
+	if scoreValue < 1 || 10 < scoreValue {
```
Comment (mbang-mlyfe) [inline slack/commands/rate.go:43]: :1st_place_medal:

File: slack/commands/rate.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+func (c *rateCommand) Execute(args string, output io.Writer) {
+	rateMatcher, err := regexp.Compile("([1-9]|10)\\s(.*)")
+	if err != nil {
+		fmt.Fprintln(output, err.Error())
+		return
+	}
+
+	rateMatches := rateMatcher.FindStringSubmatch(args)
+	scoreValue, err := strconv.ParseInt(rateMatches[1], 10, 32)
+
```
Comment (mbang-mlyfe) [inline slack/commands/rate.go:37]: Consider removing this linebreak

File: slack/commands/rate.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	score := dbctx.Score{
+		Score: int(scoreValue),
+	}
+
+	if len(args) > 2 {
+		score.Comment = rateMatches[2]
+	}
+
+	err = c.repo.SaveScore(&score)
+
```
Comment (mbang-mlyfe) [inline slack/commands/rate.go:57]: Also this one

Comment (sjuls) [reply slack/commands/rate.go]: Why? Linebreaks are like full stops - gives the developer reading it time to breathe.

File: slack/commands/rate.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
+package commands
```
Comment (mbang-mlyfe) [inline slack/commands/rate.go:1]: I quite like this command abstraction. Very nice!

File: slack/commands/rate.go (hunk 4 above)
Comment (sjuls) [reply slack/commands/rate.go:37]: Ok, I'll give you this one

File: slack/commands/rate.go (hunk 5 above)
Comment (sjuls) [reply slack/commands/rate.go:57]: I see your point of view

File: slack/commandshandler.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	if _, err := h.webAPI.PostMessage(&message); err != nil {
+		panic(err)
+	}
+}
+
+func (h *commandsHandler) Usage(output io.Writer, user string) {
+	for _, command := range h.registeredCommands {
+		if !command.RequiresAdmin() || h.isAdminUser(user) {
```
Comment (mbang-mlyfe) [inline slack/commandshandler.go:98]: No leaky! :+1:

File: soup-ranking.go [hunk 8]
```diff
@@ -4,41 +4,74 @@ import (
...
 		panic(err)
 	}
 
+	soupRepository := dbctx.NewSoupRepository(connFactory)
+	scoreRepository := dbctx.NewScoreRepository(connFactory)
+
 	router := mux.NewRouter().StrictSlash(true)
-	routes.AddStatus(router)
-	routes.AddScore(router)
+	routes := []func(router *mux.Router){
```
Comment (mbang-mlyfe) [inline soup-ranking.go:42]: I think you're going in the right direction w.r.t. dependency injection here

**Pull request merged**